qiskit_experiments/library/characterization/t1.py

```diff
@@ -82,36 +82,21 @@ def __init__(
         # Set experiment options
         self.set_experiment_options(delays=delays)
 
-    def _set_backend(self, backend: Backend):
-        super()._set_backend(backend)
-
-        # Scheduling parameters
-        if not self._backend_data.is_simulator:
-            scheduling_method = getattr(self.transpile_options, "scheduling_method", "alap")
-            self.set_transpile_options(scheduling_method=scheduling_method)
-
     def circuits(self) -> List[QuantumCircuit]:
         """
         Return a list of experiment circuits
 
         Returns:
             The experiment circuits
         """
-        dt_unit = False
-        if self.backend:
-            dt_factor = self._backend_data.dt
-            dt_unit = dt_factor is not None
+        timing = BackendTiming(self.backend)
 
         circuits = []
         for delay in self.experiment_options.delays:
```

Comment:
Probably we can refactor this to use template approach? Same for other experiments. Using parameterized template makes it easy to migrate to parameter biding with primitives in future.

Reply:
Yes, that sounds a good plan.